File: php/features.php
<?php
declare(strict_types=1);

// LaunchLibrary2 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class LaunchLibrary2Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new LaunchLibrary2BaseFeature();
            case "test":
                return new LaunchLibrary2TestFeature();
            default:
                return new LaunchLibrary2BaseFeature();
        }
    }
}

File: py/launchlibrary2_sdk/features.py
# LaunchLibrary2 SDK feature factory

from launchlibrary2_sdk.feature.base_feature import LaunchLibrary2BaseFeature
from launchlibrary2_sdk.feature.test_feature import LaunchLibrary2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: LaunchLibrary2BaseFeature(),
        "test": lambda: LaunchLibrary2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
